### PR TITLE
Documentation & tests for finding a hook using pythonpath

### LINKED ISSUE #867: Document where hooks files should go
AFAICT there's no documentation describing where hooks files should go. Worse, if you specify a hook that does not exist no error is logged to the circus log file. That's a separate bug (in 0.11.1) but if it logged the documentation might not be quite as necessary.

### PR DESCRIPTION
**Pull request by pconerly**
Documentation and a test for finding a hook using `env.pythonpath`

Resolves #867 
FYI @k4nar

### CODE REVIEW

File: docs/source/for-devs/writing-hooks.rst [hunk 1]
```diff
@@ -89,6 +89,21 @@ Just before starting the second watcher, it will run the **check_redis**
...
+PYTHONPATH in *[env:watchername]*.
+
+.. code-block:: ini
+
+    [watcher:foo]
+    copy_env = True
+    hooks.before_start = hooks.my_hook.hook
+
+    [env:foo]
+    PYTHONPATH = $PWD/circus/tests/config
```
Comment (k4nar) [inline docs/source/for-devs/writing-hooks.rst]: I think in this example we should use `PYTHONPATH=$PYTHONPATH:$PWD` instead

File: docs/source/for-devs/writing-hooks.rst [hunk 2]
```diff
@@ -89,6 +89,21 @@ Just before starting the second watcher, it will run the **check_redis**
...
+
+You can use bash commands like *$PWD* in the *PYTHONPATH*.
```
Comment (k4nar) [inline docs/source/for-devs/writing-hooks.rst]: s/bash commands/env variables/ :) .

**Comment by k4nar** [comment]
Thanks! I've made a few comments, but otherwise it's really great to have an exemple in the docs.

**Comment by pconerly** [comment]
I amended the commit to remove the circus/tests/config

**Pull request merged**